src/NativeAdComponents.js

```diff
@@ -7,20 +7,21 @@ export const TitleView = (props) => {
   const {nativeAd, nativeAdView} = useContext(NativeAdViewContext);
 
   useEffect(() => {
-    if (Object.keys(nativeAd).length === 0 || !nativeAdView) return;
+    if (!nativeAdView) return;
+
+    if (!nativeAd.title) return;
```

Comment:
nit - AppLovin coding convention we try to return early for ease of readability + concise.
```suggestion
    if (!nativeAdView || !nativeAd.title) return;
```

Reply:
Also, is the check for `!nativeAdView` necessary? Since we have it on L19 already

Reply:
Comments here apply for other parts of this file (where applicable).

Reply:
The 2nd `!nativeAdView` checking is a bit different and necessary.    `nativeAdView` is the parent view.   When the parent is created, this child view needs to be created to get its `viewRef` before `nativeAd` is not set by the platform side.   Then, when`nativeAd` becomes available, things can be set via `viewRef`.